packages/astro/src/core/render-context.ts

```diff
@@ -314,8 +317,10 @@ export class RenderContext {
 			renderers,
 			resolve,
 			response,
+			request: this.request.clone(),
```

Comment:
Can this be done conditionally only when necessary? Cloning every request would be expensive.

Reply:
I have a feeling this is slowing the test suite too. The clone() was a precaution but it shouldn't be necessary? I'll try just removing the clone(). I wouldn't know a conditional to check 🤷

Reply:
If that doesn't work, an alternative would be to _just_ pass `requestFormData` when a form data content type is present. This would scope down to the info we actually need to access from the React integration.

Reply:
Confirmed that clone was the slowdown. Good to know the impact that has